hatch/src/cli/commands/new.rs

```diff
@@ -53,3 +61,30 @@ impl<'command> Command<'command> for New {
         Ok(())
     }
 }
+
+
+#[cfg(test)]
+mod tests {
+    #[test]
+    fn it_makes_correct_dirs() {
+        use super::New;
+        use assert_fs::TempDir;
+        use assert_fs::prelude::*;
+        use predicates::prelude::*;
+        let temp = TempDir::new().unwrap();
+        let dir_path = temp.path().to_owned();
+        let new = New::new();
+        let result = new.write_directory_structure(&dir_path);
+        let is_dir = predicate::path::is_dir();
+        temp.child("src").assert(&is_dir);
```

Comment:
This is really neat!  So is "is_dir" a function? And assert can take a function?

Reply:
Correct.

Reply:
the predicates library might be useful for other things too, it is not specific to assert_fs

Reply:
That's awesome and definitely!